### PR TITLE
context: cleanup paths-not-taken

### PR DESCRIPTION
**Pull request by efritz**
This PR undoes changes from #51690 that are no longer necessary (exporting functions still unused outside of their package, methods split into chunks we no longer need, etc). Merging this will make the diff against main much smaller.

## Test plan

Tested locally.

### CODE REVIEW

File: client/cody-shared/src/chat/recipes/helpers.ts [hunk 1]
```diff
@@ -70,41 +70,3 @@ export function contentSanitizer(text: string): string {
     }
     return output.replace(/^\s*\n/, '')
 }
-
-export function convertGitCloneURLToCodebaseName(cloneURL: string): string | null {
```
Comment (efritz) [inline client/cody-shared/src/chat/recipes/helpers.ts:74]: Unused (copied from another source, but we import _that_ one still).

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 2]
```diff
@@ -12,97 +11,28 @@ import (
...
 
 	"github.com/sourcegraph/sourcegraph/internal/codeintel/codenav/shared"
-	"github.com/sourcegraph/sourcegraph/internal/codeintel/shared/symbols"
 	"github.com/sourcegraph/sourcegraph/internal/observation"
 	"github.com/sourcegraph/sourcegraph/lib/codeintel/precise"
 )
 
 // GetDefinitionLocations returns the set of locations defining the symbol at the given position.
 func (s *store) GetDefinitionLocations(ctx context.Context, bundleID int, path string, line, character, limit, offset int) (_ []shared.Location, _ int, err error) {
-	ctx, trace, endObservation := s.operations.getDefinitionLocations.With(ctx, &err, observation.Args{Attrs: []attribute.KeyValue{
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:22]: We merged codenav changes from #53710, making this split unnecessary. Just undoing it here.

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 3]
```diff
@@ -127,24 +57,13 @@ func (s *store) GetBulkMonikerLocations(ctx context.Context, tableName string, u
 	explodedSymbols := []string{}
 	symbolNames := make([]string, 0, len(monikers))
 	for _, arg := range monikers {
-		symbolNames = append(symbolNames, arg.Identifier)
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:130]: Use the `explodeSymbol` helper for convenience (and to DRY the encoding so if we change it it's localized).

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 4]
```diff
@@ -214,68 +133,72 @@ JOIN codeintel_scip_document_lookup dl ON dl.id = ss.document_lookup_id
...
+	occurrences := scip.FindOccurrences(documentData.SCIPData.Occurrences, int32(line), int32(character))
+	trace.AddEvent("FindOccurences", attribute.Int("numIntersectingOccurrences", len(occurrences)))
 
 	for _, occurrence := range occurrences {
 		var locations []shared.Location
 		if ranges := scipExtractor(documentData.SCIPData, occurrence); len(ranges) != 0 {
 			locations = append(locations, convertSCIPRangesToLocations(ranges, bundleID, path)...)
 		}
 
-		locs, err := s.GetLocationByExplodedSymbol(ctx, occurrence.Symbol, bundleID, scipFieldName, path)
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:273]: Re-inlining `GetLocationByExplodedSymbol`, which we don't need to expose anymore.

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 5]
```diff
@@ -214,68 +133,72 @@ JOIN codeintel_scip_document_lookup dl ON dl.id = ss.document_lookup_id
...
-		if err != nil {
-			return nil, 0, err
-		}
+		if occurrence.Symbol != "" && !scip.IsLocalSymbol(occurrence.Symbol) {
+			explodedSymbol, err := explodeSymbol(occurrence.Symbol)
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:173]: Use the `explodeSymbol` helper for convenience (and to DRY the encoding so if we change it it's localized).

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 6]
```diff
@@ -297,55 +220,6 @@ func (s *store) getLocations(
...
-	uploadID int,
-	scipFieldName string,
-	path string,
-) (locations []shared.Location, err error) {
-	if symbolName != "" && !scip.IsLocalSymbol(symbolName) {
-		ex, err := symbols.NewExplodedSymbol(symbolName)
-		if err != nil {
-			return nil, err
-		}
-		explodedSymbols := fmt.Sprintf(
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:312]: Use the `explodeSymbol` helper for convenience (and to DRY the encoding so if we change it it's localized).

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 7]
```diff
@@ -127,24 +57,13 @@ func (s *store) GetBulkMonikerLocations(ctx context.Context, tableName string, u
...
 	for _, arg := range monikers {
-		symbolNames = append(symbolNames, arg.Identifier)
-
-		// fmt.Println("args.Identifier: ", arg.Identifier)
-		s, err := symbols.NewExplodedSymbol(arg.Identifier)
+		explodedSymbol, err := explodeSymbol(arg.Identifier)
 		if err != nil {
 			return nil, 0, err
 		}
-		explodedSymbols = append(
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:137]: Use the `explodeSymbol` helper for convenience (and to DRY the encoding so if we change it it's localized).

File: internal/codeintel/codenav/internal/lsifstore/locations_by_position.go [hunk 8]
```diff
@@ -668,6 +549,8 @@ func (s *store) GetMinimalBulkMonikerLocations(ctx context.Context, tableName st
 		minimalBulkMonikerResultsQuery,
 		pq.Array(symbolNames),
 		pq.Array(uploadIDs),
+		pq.Array(explodedSymbolNames),
+		pq.Array(uploadIDs),
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/locations_by_position.go:553]: We were missing parameters here (drive-by bug fix).

File: internal/codeintel/codenav/internal/lsifstore/lsifstore_documents.go [hunk 9]
```diff
@@ -64,25 +63,11 @@ func (s *store) GetFullSCIPNameByDescriptor(ctx context.Context, uploadID []int,
...
-		var n types.SCIPNames
-		if err := rows.Scan(&n.Scheme, &n.PackageManager, &n.PackageName, &n.PackageVersion, &n.Descriptor); err != nil {
-			return nil, err
-		}
-
-		names = append(names, &n)
-	}
-
-	return names, nil
+	return scanExplodedSymbols(s.db.Query(ctx, sqlf.Sprintf(
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/lsifstore_documents.go:66]: Wrote a scanner for this

File: internal/codeintel/codenav/internal/lsifstore/util.go [hunk 10]
```diff
@@ -22,3 +26,19 @@ func newRange(startLine, startCharacter, endLine, endCharacter int) shared.Range
 		},
 	}
 }
+
+func explodeSymbol(symbol string) (string, error) {
```
Comment (efritz) [inline internal/codeintel/codenav/internal/lsifstore/util.go:30]: so dryyyy

File: internal/codeintel/shared/symbols/symbols.go [hunk 11]
```diff
@@ -10,7 +12,29 @@ type ExplodedSymbol struct {
...
 	Descriptor         string
-	DescriptorNoSuffix string
+	DescriptorNoSuffix string // N.B.
+}
+
+func (s *ExplodedSymbol) String() string {
+	return s.Symbol()
+}
+
+func (s *ExplodedSymbol) Symbol() string {
```
Comment (efritz) [inline internal/codeintel/shared/symbols/symbols.go:22]: Takes the place of `SCIPNames.GetIdentifier`.

File: internal/codeintel/types/context.go [hunk 12]
```diff
@@ -1,41 +1,6 @@
...
-	}
-	descriptor := "."
-	if s.Descriptor != "" {
-		descriptor = s.Descriptor
-	}
-	return fmt.Sprintf("%s %s %s %s %s", scheme, manager, name, version, descriptor)
-}
-
-type PreciseData struct {
+type PreciseContext struct {
```
Comment (efritz) [inline internal/codeintel/types/context.go:3]: smol rename

File: internal/codeintel/context/service.go [hunk 13]
```diff
@@ -172,27 +168,23 @@ func (s *Service) GetPreciseContext(ctx context.Context, args *resolverstubs.Get
...
-					return nil, err
-				}
-
-				preciseDataList = append(preciseDataList, &preciseData{
-					syntectName: syntectName,
-					symbolName:  ident,
-					location:    ul,
-				})
+			// for _, upload := range uploads {
+			ul, err := s.codenavSvc.NewGetDefinitionsBySymbolNames(ctx, requestArgs, reqState, []string{ident})
```
Comment (efritz) [inline internal/codeintel/context/service.go:177]: Use real codenav method (from #53710).

**Comment by efritz** [comment]
Continue in #54693.

**Review by Numbers88s — APPROVED** [review]
